server/driver/wivrn_session.cpp

```diff
@@ -304,6 +304,11 @@ void wivrn_session::operator()(from_headset::feedback && feedback)
 		dump_time("display", feedback.frame_index, o.from_headset(feedback.displayed), feedback.stream_index);
 }
 
+void wivrn_session::operator()(from_headset::battery && battery)
+{
+	hmd->update_battery(battery);
```

Comment:
We need some locking, this is called from network thread but reads are from monado thread

Reply:
just `std::lock_guard lock(mutex);` is good enough?

Reply:
The comment should be in `update_battery`, use the mutex in `wivrn_hmd` and it's fine (for read too of course).